File: src/test/java/com/raychenon/leetcode/NonOverlappingIntervalsTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 2/9/2020
 * https://leetcode.com/problems/non-overlapping-intervals/
 */
public class NonOverlappingIntervalsTest {

    private static NonOverlappingIntervals instance;

    @BeforeClass
    public static void setUp() {
        instance = new NonOverlappingIntervals();
    }

    @Test
    public void example1Test() {
        int[][] input = {{1, 2}, {2, 3}, {3, 4}, {1, 3}};
        assertNonOverlappingIntervals(1, input);
    }

    @Test
    public void example2Test() {
        int[][] input = {{1, 2}, {1, 2}, {1, 2}};
        assertNonOverlappingIntervals(2, input);
    }

    @Test
    public void example3Test() {
        int[][] input = {{1, 2}, {2, 3}};
        assertNonOverlappingIntervals(0, input);
    }

    @Test
    public void example4Test() {
        int[][] array = {{1, 100}, {11, 22}, {1, 11}, {2, 12}};
        assertNonOverlappingIntervals(2, array);
    }

    private void assertNonOverlappingIntervals(int expected, int[][] arr) {
        assertEquals(expected, instance.eraseOverlapIntervals(arr));
    }

}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/NonOverlappingIntervalsTest.kt
package com.raychenon.kotlin.leetcode

import junit.framework.TestCase
import org.junit.Test

/**
 * User: raychenon
 * Date: 2/9/2020
 * https://leetcode.com/problems/non-overlapping-intervals/
 */
class NonOverlappingIntervalsTest {

    @Test
    fun example1Test() {
        val input = arrayOf(intArrayOf(1, 2), intArrayOf(2, 3), intArrayOf(3, 4), intArrayOf(1, 3))
        assertNonOverlappingIntervals(1, input)
    }

    @Test
    fun example2Test() {
        val input = arrayOf(intArrayOf(1, 2), intArrayOf(1, 2), intArrayOf(1, 2))
        assertNonOverlappingIntervals(2, input)
    }

    @Test
    fun example3Test() {
        val input = arrayOf(intArrayOf(1, 2), intArrayOf(2, 3))
        assertNonOverlappingIntervals(0, input)
    }

    @Test
    fun example4Test() {
        val array = arrayOf(intArrayOf(1, 100), intArrayOf(11, 22), intArrayOf(1, 11), intArrayOf(2, 12))
        assertNonOverlappingIntervals(2, array)
    }

    private fun assertNonOverlappingIntervals(expected: Int, arr: Array<IntArray>) {
        TestCase.assertEquals(expected, NonOverlappingIntervals.eraseOverlapIntervals(arr))
    }

}